### ISSUE TITLE
Single Build for Data Prepper

### ISSUE THREAD

**Issue by dlvenable** [opensearch-project/opensearch-build#1465]
### Is your feature request related to a problem? Please describe

Data Prepper releases uses two custom builds for the tarballs and a parameterized build for the Docker image.

### Describe the solution you'd like

One Data Prepper build which produces three artifact sets:

* Docker image
* Tarball files
* Maven artifacts

It should also handle promotion.

This single job should merge the work done in #1434 and #1460.

### Additional context

GitHub Issues to complete this task:

* [x] #1594 
* [x] #1751 
* [x] #1737

**Comment by abhinavGupta16** [comment]
This task would also require us to remove the files merged in https://github.com/opensearch-project/opensearch-build/pull/1460. 

Following approach can be used to promote artifact - https://github.com/opensearch-project/opensearch-build/pull/1460#pullrequestreview-851619395

Can this be added in the description?

**Comment by dlvenable** [comment]
We are currently working on this Issue and I have added three sub-tasks to the description.

We have already created a new job which promotes tar.gz archives and the Docker images. This was the first part of this new unified single build. The next task is to support Maven promotion to Sonatype.

After our first release using this process (Data Prepper 1.3.0), we will remove the old files. That should complete this initiative.
